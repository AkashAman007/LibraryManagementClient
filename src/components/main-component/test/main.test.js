import React from 'react';
import {render, cleanup, wait } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import  Main  from "../main";
import { MemoryRouter } from 'react-router-dom';

afterEach(cleanup);

it(" renders main with book listing page", async () => {
    await wait();
    const tree = render(<MemoryRouter initialEntries={["/"]}><Main/></MemoryRouter>)
    expect(tree.asFragment()).toMatchSnapshot();
});

it(" renders main with user book listing page", async () => {
    await wait();
    const tree = render(<MemoryRouter initialEntries={["/MyBooks"]}><Main/></MemoryRouter>)
    expect(tree.asFragment()).toMatchSnapshot();
});
