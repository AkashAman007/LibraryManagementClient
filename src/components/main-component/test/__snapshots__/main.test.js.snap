// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` renders main with book listing page 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        <h1
          class="title"
        >
          Available Books in Library
          <a
            class="float-right link"
            data-testid="my-books-link"
            href="/MyBooks"
          >
            See Borrowed Books
          </a>
        </h1>
        <div
          class="info"
        >
          <div />
        </div>
        <div
          class="book-grid"
        >
          <div>
            No books in library
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[` renders main with user book listing page 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        <h1
          class="title"
        >
          My Books
          <a
            class="float-right link"
            data-testid="library-books-link"
            href="/"
          >
            View Books
          </a>
        </h1>
        <div
          class="info"
        >
          <div />
        </div>
        <div
          class="no-content"
        >
          You don't have any borrowed book
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
